domain/consensus/datastructures/utxodiffstore/utxodiffstore.go

```diff
@@ -3,45 +3,119 @@ package utxodiffstore
+var utxoDiffChildBucket = dbkeys.MakeBucket([]byte("utxo-diff-children"))
+
 // utxoDiffStore represents a store of UTXODiffs
 type utxoDiffStore struct {
+	utxoDiffStaging      map[externalapi.DomainHash]*model.UTXODiff
+	utxoDiffChildStaging map[externalapi.DomainHash]*externalapi.DomainHash
 }
 
 // New instantiates a new UTXODiffStore
 func New() model.UTXODiffStore {
-	return &utxoDiffStore{}
+	return &utxoDiffStore{
+		utxoDiffStaging:      make(map[externalapi.DomainHash]*model.UTXODiff),
+		utxoDiffChildStaging: make(map[externalapi.DomainHash]*externalapi.DomainHash),
+	}
 }
 
 // Stage stages the given utxoDiff for the given blockHash
 func (uds *utxoDiffStore) Stage(blockHash *externalapi.DomainHash, utxoDiff *model.UTXODiff, utxoDiffChild *externalapi.DomainHash) {
-	panic("implement me")
+	uds.utxoDiffStaging[*blockHash] = utxoDiff
+	uds.utxoDiffChildStaging[*blockHash] = utxoDiffChild
 }
 
 func (uds *utxoDiffStore) IsStaged() bool {
-	panic("implement me")
+	return len(uds.utxoDiffStaging) != 0
 }
 
 func (uds *utxoDiffStore) Discard() {
-	panic("implement me")
+	uds.utxoDiffStaging = make(map[externalapi.DomainHash]*model.UTXODiff)
+	uds.utxoDiffChildStaging = make(map[externalapi.DomainHash]*externalapi.DomainHash)
 }
 
 func (uds *utxoDiffStore) Commit(dbTx model.DBTransaction) error {
-	panic("implement me")
+	for hash, utxoDiff := range uds.utxoDiffStaging {
+		err := dbTx.Put(uds.utxoDiffHashAsKey(&hash), uds.serializeUTXODiff(utxoDiff))
+		if err != nil {
+			return err
+		}
+	}
+	for hash, utxoDiffChild := range uds.utxoDiffChildStaging {
+		err := dbTx.Put(uds.utxoDiffHashAsKey(&hash), utxoDiffChild[:])
+		if err != nil {
+			return err
+		}
+	}
+
+	uds.Discard()
+	return nil
 }
 
 // UTXODiff gets the utxoDiff associated with the given blockHash
 func (uds *utxoDiffStore) UTXODiff(dbContext model.DBReader, blockHash *externalapi.DomainHash) (*model.UTXODiff, error) {
-	return nil, nil
+	if utxoDiff, ok := uds.utxoDiffStaging[*blockHash]; ok {
+		return utxoDiff, nil
+	}
+
+	utxoDiffBytes, err := dbContext.Get(uds.utxoDiffHashAsKey(blockHash))
+	if err != nil {
+		return nil, err
+	}
+
+	return uds.deserializeUTXODiff(utxoDiffBytes)
 }
 
 // UTXODiffChild gets the utxoDiff child associated with the given blockHash
 func (uds *utxoDiffStore) UTXODiffChild(dbContext model.DBReader, blockHash *externalapi.DomainHash) (*externalapi.DomainHash, error) {
-	return nil, nil
+	if utxoDiffChild, ok := uds.utxoDiffChildStaging[*blockHash]; ok {
+		return utxoDiffChild, nil
+	}
+
+	utxoDiffChildBytes, err := dbContext.Get(uds.utxoDiffChildHashAsKey(blockHash))
+	if err != nil {
+		return nil, err
+	}
+	var utxoDiffChild externalapi.DomainHash
+	copy(utxoDiffChild[:], utxoDiffChildBytes[:])
+
+	return &utxoDiffChild, nil
 }
 
 // Delete deletes the utxoDiff associated with the given blockHash
 func (uds *utxoDiffStore) Delete(dbTx model.DBTransaction, blockHash *externalapi.DomainHash) error {
-	return nil
+	if uds.IsStaged() {
```

Comment:
You need to check if one of them is staged instead of checking if there's anything in staging